File: src/main/java-graphql/io/github/graphql/model/PullRequestReviewEdgeTO.java
package io.github.graphql.model;

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer;
import java.util.StringJoiner;

@javax.annotation.Generated(
    value = "com.kobylynskyi.graphql.codegen.GraphQLCodegen",
    date = "2021-03-06T19:47:47+0800"
)
public class PullRequestReviewEdgeTO implements java.io.Serializable {

    @javax.validation.constraints.NotNull
    private String cursor;
    private PullRequestReviewTO node;

    public PullRequestReviewEdgeTO() {
    }

    public PullRequestReviewEdgeTO(String cursor, PullRequestReviewTO node) {
        this.cursor = cursor;
        this.node = node;
    }

    public String getCursor() {
        return cursor;
    }

    public PullRequestReviewTO getNode() {
        return node;
    }


    @Override
    public String toString() {
        StringJoiner joiner = new StringJoiner(", ", "{ ", " }");
        if (cursor != null) {
            joiner.add("cursor: " + GraphQLRequestSerializer.getEntry(cursor));
        }
        if (node != null) {
            joiner.add("node: " + GraphQLRequestSerializer.getEntry(node));
        }
        return joiner.toString();
    }

    public static PullRequestReviewEdgeTO.Builder builder() {
        return new PullRequestReviewEdgeTO.Builder();
    }

    public static class Builder {

        private String cursor;
        private PullRequestReviewTO node;

        public Builder() {
        }

        public Builder setCursor(String cursor) {
            this.cursor = cursor;
            return this;
        }

        public Builder setNode(PullRequestReviewTO node) {
            this.node = node;
            return this;
        }


        public PullRequestReviewEdgeTO build() {
            return new PullRequestReviewEdgeTO(cursor, node);
        }

    }
}

File: src/main/kotlin-graphql/io/github/graphql/model/PullRequestReviewEdgeTO.kt
package io.github.graphql.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer
import java.util.StringJoiner

@javax.annotation.Generated(
    value = ["com.kobylynskyi.graphql.codegen.GraphQLCodegen"],
    date = "2021-03-06T19:48:03+0800"
)
open class PullRequestReviewEdgeTO(
    val cursor: String,
    val node: PullRequestReviewTO?
) {

    companion object {
        fun builder(): Builder = Builder()
    }

    // In the future, it maybe change.
    override fun toString(): String {
        val joiner = StringJoiner(", ", "{ ", " }")
        joiner.add("cursor: " + GraphQLRequestSerializer.getEntry(cursor))
        if (node != null) {
            joiner.add("node: " + GraphQLRequestSerializer.getEntry(node))
        }
        return joiner.toString()
    }

    class Builder {

        private lateinit var cursor: String
        private var node: PullRequestReviewTO? = null

        fun setCursor(cursor: String): Builder {
            this.cursor = cursor
            return this
        }

        fun setNode(node: PullRequestReviewTO?): Builder {
            this.node = node
            return this
        }

        fun build(): PullRequestReviewEdgeTO {
            return PullRequestReviewEdgeTO(cursor, node)
        }
    }
}
